pkg/yang/file.go

```diff
@@ -115,22 +124,43 @@ func findFile(name string) (string, string, error) {
 	return "", "", fmt.Errorf("no such file: %s", name)
 }
 
-// findInDir looks for a file named name in dir or any of its subdirectories.
-func findInDir(dir, name string) string {
+// findInDir looks for a file named name in dir or any of its subdirectories if
+// recurse is true. if recurse is false, scan only the directory dir.
+func findInDir(dir, name string, recurse bool) string {
 	fis, err := ioutil.ReadDir(dir)
 	if err != nil {
 		return ""
 	}
+	var candidates []string
+	mname := name
+	if strings.HasSuffix(mname, ".yang") {
+		mname = name[:len(name)-len(".yang")]
+	}
+
 	for _, fi := range fis {
 		if !fi.IsDir() {
 			if fi.Name() == name {
 				return filepath.Join(dir, name)
+			} else if !strings.Contains(name, "@") && strings.HasPrefix(fi.Name(), mname+"@") && strings.HasSuffix(fi.Name(), ".yang") {
```

Comment:
I think you can drop the strings.Contains as it  is implicit with the HasPrefix call.

If you are worried about the call to fi.Name(), you could make a local variable and use it throughout this block (below line 141) but I don't think that it is really necessary.